man/man1/lazyman.1
.\" Automatically generated by Pandoc 2.19.2
.\"
.\" Define V font for inline verbatim, using C font in formats
.\" that render this, and otherwise B font.
.ie "\f[CB]x\f[]"x" \{\
. ftr V B
. ftr VI BI
. ftr VB B
. ftr VBI BI
.\}
.el \{\
. ftr V CR
. ftr VI CI
. ftr VB CB
. ftr VBI CBI
.\}
.TH "LAZYMAN" "1" "March 13, 2023" "lazyman 2.1.8" "User Manual"
.hy
.SH LAZYMAN
.SS NAME
.PP
lazyman - install, initialize, manage, and explore multiple Neovim
configurations
.SS SYNOPSIS
.PP
lazyman [-4] [-9] [-A] [-a] [-B] [-b branch] [-c] [-d] [-E config] [-e]
[-f path] [-F menu] [-g] [-i group] [-j] [-k] [-l] [-m] [-M] [-s] [-S]
[-v] [-n] [-o] [-O name] [-p] [-P] [-q] [-Q] [-h] [-H] [-I] [-J] [-L
lang] [-rR] [-C url] [-D subdir] [-N nvimdir] [-G] [-tT] [-U] [-V url]
[-w conf] [-W] [-x conf] [-X] [-y] [-Y] [-z] [-Z] [-K conf] [-u]
[health] [info] [init] [install [bob]] [open] [remove] [search] [status]
[usage]
.PP
Visit the Lazyman website (https://lazyman.dev) for more info.
.SS FEATURES
.IP \[bu] 2
\f[V]lazyman\f[R] command to easily install, initialize, manage, and
explore multiple Neovim configurations
.IP \[bu] 2
support for \f[V]Lazy\f[R], \f[V]Packer\f[R], and \f[V]vim-plug\f[R]
plugin managers (\f[V]dein\f[R] for \f[V]SpaceVim\f[R] only)
.IP \[bu] 2
open, install, remove, get info, search plugins, all from the command
line or main menu
.IP \[bu] 2
automated installation of dependencies, tools, language servers, and
Neovim 0.9+
.IP \[bu] 2
install and manage the \f[V]Bob\f[R] neovim version manager via a menu
interface
.IP \[bu] 2
richly configured \f[V]nvim-Lazyman\f[R] Neovim configuration
.IP \[bu] 2
interactive menu interface for ease of management
.RS 2
.IP \[bu] 2
manage the \f[V]nvim-Lazyman\f[R], \f[V]nvim-LazyIde\f[R], and
\f[V]nvim-Webdev\f[R] configs via menus
.IP \[bu] 2
perform health checks and generate a status report via menus
.RE
.IP \[bu] 2
over 100 supported Neovim configurations out of the box, additional
custom configs
.IP \[bu] 2
vimdoc help for \f[V]Lazyman\f[R] with \f[V]:h Lazyman\f[R] or
\f[V],hl\f[R]
.IP \[bu] 2
convenience shell functions and aliases with fuzzy search and selection
.RS 2
.IP \[bu] 2
\f[V]nvims\f[R] and \f[V]neovides\f[R] shell functions to fuzzy search,
select, and open Neovim configs
.IP \[bu] 2
enhanced \f[V]less\f[R] command alias
.IP \[bu] 2
enhanced \f[V]ls\f[R] command alias
.IP \[bu] 2
\f[V]tree\f[R] alias to display a tree view of files and folders
.IP \[bu] 2
\f[V]tldrf\f[R] alias to fuzzy search, select, and preview cheat sheets
.RE
.PP
See the Lazyman website Features section (https://lazyman.dev/features)
for additional information on Lazyman features.
.SS DESCRIPTION
.PP
The \f[V]lazyman\f[R] command can be used to install, initialize, and
manage multiple Neovim configurations.
Neovim configurations are installed and initialized in separate
subdirectories of \f[V]\[ti]/.config/\f[R] with associated separate
subdirectories in \f[V]\[ti]/.local/share/\f[R],
\f[V]\[ti]/.local/state/\f[R], and \f[V]\[ti]/.cache/\f[R].
.PP
Note that a full installation and initialization of all supported Neovim
configurations, plugins, language servers, formatters, linters, and
tools will consume over 20GB of disk space.
.PP
The \f[V]lazyman\f[R] Neovim configuration is installed as
\f[V]\[ti]/.config/nvim-Lazyman\f[R] and the \f[V]lazyman\f[R] command
is installed as \f[V]\[ti]/.local/bin/lazyman\f[R].
.PP
The Neovim configuration used by the \f[V]nvim\f[R] command is
determined by the \f[V]NVIM_APPNAME\f[R] environment variable (Neovim
0.9 and later only).
Set the \f[V]NVIM_APPNAME\f[R] environment variable in the shell
initialization file (typically \f[V]\[ti]/.bashrc\f[R] or
\f[V]\[ti]/.zshrc\f[R]) by using the \f[V]export\f[R] shell builtin:
.IP
.nf
\f[C]
export NVIM_APPNAME=\[dq]<nvimdir>\[dq]
\f[R]
.fi
.PP
where \f[V]<nvimdir>\f[R] is the subdirectory in
\f[V]\[ti]/.config/\f[R] that holds the desired Neovim configuration.
For example, to use the \f[V]LazyVim\f[R] Neovim configuration in
\f[V]\[ti]/.config/nvim-LazyVim/\f[R] the export command would be:
.IP
.nf
\f[C]
export NVIM_APPNAME=\[dq]nvim-LazyVim\[dq]
\f[R]
.fi
.PP
In addition to exporting NVIM_APPNAME in your shell initialization file,
you may wish to create aliases to execute with the various Neovim
configurations you have installed.
For example, aliases could be created to use Neovim configurations
installed in \f[V]\[ti]/.config/nvim-LazyVim\f[R] and
\f[V]\[ti]/.config/nvim-LunarVim\f[R] as follows:
.IP
.nf
\f[C]
alias nvim-lazy=\[dq]NVIM_APPNAME=nvim-LazyVim nvim\[dq]
alias nvim-lunar=\[dq]NVIM_APPNAME=nvim-LunarVim nvim\[dq]
\f[R]
.fi
.PP
After sourcing these aliases in your shell, to invoke Neovim with the
LazyVim configuration run \f[V]nvim-lazy filename.py\f[R] and to invoke
Neovim with the LunarVim config run \f[V]nvim-lunar proposal.md\f[R].
.PP
The \f[V]lazyman\f[R] installation and configuration automatically
configures convenience aliases and shell functions for Lazyman installed
Neovim configurations.
It also creates an \f[V]nvims\f[R] shell function which dynamically
creates a fuzzy searchable menu of installed Neovim configurations and
launches Neovim with the selected Lazyman Neovim configuration.
.PP
See \f[V]\[ti]/.config/nvim-Lazyman/.lazymanrc\f[R].
.PP
Similarly, a \f[V]neovides\f[R] shell function can be used to select a
Neovim configuration for use with the Neovim GUI \f[V]neovide\f[R].
.PP
Both the \f[V]nvims\f[R] shell function and \f[V]neovides\f[R] shell
function accept command line arguments.
The \f[V]-R\f[R] flag indicates removal of the selected Neovim
configuration.
The \f[V]-C filter\f[R] option to \f[V]nvims\f[R] and \f[V]neovides\f[R]
specifies a filter string to match when generating the list of Neovim
configurations to search and select.
The \f[V]-I\f[R] flag indicates display the selected configuration
information document.
Additional command line arguments are described below.
.PP
The fuzzy searchable/selectable menu of Neovim configurations can also
be shown with the command \f[V]lazyman -S\f[R].
Note also that both the \f[V]nvims\f[R] shell function and the
\f[V]lazyman -S\f[R] command can accept additional filename arguments
which are then passed to Neovim.
For example, to edit \f[V]/tmp/foo.lua\f[R] with a Neovim configuration
selected from the \f[V]nvims\f[R] menu:
.IP
.nf
\f[C]
nvims /tmp/foo.lua
\f[R]
.fi
.PP
Execute \f[V]nvims\f[R] directly at the shell prompt or by using the
convenience key binding \f[V]ctrl-n\f[R].
.PP
Similarly, if \f[V]neovide\f[R] is found in the execution PATH then a
fuzzy selectable menu is provided with the \f[V]neovides\f[R] shell
function and convenience key binding of \f[V]ctrl-N\f[R] to bring up
that menu.
.PP
The \f[V].lazymanrc\f[R] file also creates aliases for \f[V]ls\f[R],
\f[V]less\f[R], and others including a \f[V]tldrf\f[R] alias that allows
fuzzy searching and selecting from the list of cheatsheets supported by
\f[V]tldr\f[R].
Try out the \f[V]tldrf\f[R] command to quickly and easily display
documentation for thousands of commands.
.PP
Another alternative to setting \f[V]NVIM_APPNAME\f[R] in the environment
or with an alias is to use the \f[V]lazyman\f[R] command to specify
which Neovim configuration to use with this invocation.
This is done using the \f[V]-E config\f[R] option to \f[V]lazyman\f[R].
When invoking \f[V]lazyman\f[R] with the \f[V]-E config\f[R] argument,
the Neovim configuration can be specified by setting \f[V]config\f[R] to
one of \f[V]astronvim\f[R], \f[V]ecovim\f[R], \f[V]kickstart\f[R],
\f[V]lazyman\f[R], \f[V]lazyvim\f[R], \f[V]lunarvim\f[R],
\f[V]nvchad\f[R], \f[V]spacevim\f[R], or any Neovim configuration
directory in \f[V]\[ti]/.config\f[R].
For example, to edit the file \f[V]foo.lua\f[R] using the LazyVim Neovim
configuration:
.IP
.nf
\f[C]
lazyman -E lazyvim foo.lua
\f[R]
.fi
.PP
When invoked with the \f[V]-E config\f[R] option, \f[V]lazyman\f[R] sets
the \f[V]NVIM_APPNAME\f[R] environment variable to the specified
\f[V]config\f[R] and executes \f[V]nvim\f[R] with all following
arguments.
This is a pretty easy way to explore all the \f[V]lazyman\f[R] installed
and initialized Neovim configurations.
.SS Supported plugin managers
.PP
Lazyman currently supports the following Neovim plugin managers:
.IP \[bu] 2
Lazy (https://github.com/folke/lazy.nvim) (lazy.nvim)
.IP \[bu] 2
Packer (https://github.com/wbthomason/packer.nvim) (packer.nvim)
.IP \[bu] 2
Plug (https://github.com/junegunn/vim-plug) (vim-plug)
.PP
The SpaceVim bundled plugin manager is also supported.
.PP
Neovim configurations using other plugin managers will likely fail to
cleanly install and initialize using \f[V]lazyman\f[R].
Support for additional plugin managers is not currently planned but if
you have a need for this feature open an issue.
.PP
To install and initialize a Neovim configuration that uses the
\f[B]Packer\f[R] plugin manager invoke \f[V]lazyman\f[R] with the
\f[V]-P\f[R] flag.
.PP
To install and initialize a Neovim configuration that uses the
\f[B]Plug\f[R] plugin manager invoke \f[V]lazyman\f[R] with the
\f[V]-p\f[R] flag.
.SS Lazyman menu system
.PP
The \f[V]lazyman\f[R] command, when invoked without arguments or with
the \f[V]-F menu\f[R] argument, presents an interactive menu.
The Lazyman menu system can be used to perform almost all of the
functions available from the command line.
Submenus are available to configure the Lazyman Neovim configuration as
well as the \f[V]Webdev\f[R] and \f[V]LazyIde\f[R] Neovim
configurations.
.PP
Use the Lazyman menu system to select and install/open/remove Neovim
configurations managed by Lazyman.
Enter a menu option number or keywords to select an option.
.PP
Keywords include: help, info, install, open, remove, search, update,
quit
.PP
All shortcuts have single key support:
.IP
.nf
\f[C]
  \[aq]h\[aq] = help, \[aq]I\[aq] = info, \[aq]i\[aq] = install, \[aq]o\[aq] = open, \[aq]q\[aq] = quit
  \[aq]r\[aq] = remove, \[aq]s\[aq] = search, \[aq]u\[aq] = update, \[aq]c\[aq] = Lazyman Config menu
\f[R]
.fi
.PP
In the fuzzy selection dialogs, enter a few letters to fuzzy select from
the options or use the \f[V]<Up-Arrow>\f[R] and \f[V]<Down-Arrow>\f[R]
keys to move through the options.
Press \f[V]<Enter>\f[R] to select the highlighted option.
.PP
Lazyman Neovim configurations are organized into categories.
Configurations can be individually installed/removed or all
configurations in a category can be acted on.
.PP
The main menu also provides options to install additional tools, enable
debug mode, install the \f[V]Bob\f[R] Neovim version manager, select the
Neovim version to use, toggle the user interface between
\f[V]Neovim\f[R] and \f[V]Neovide\f[R], perform a Neovim health check,
bring up the Lazyman Neovim Configuration menu, or generate a status
report.
In addition, an option exists to view the Lazyman manual.
.PP
The Lazyman Neovim configuration manager can also be used from the
command line.
.SS OPTIONS
.PP
The following command line options are available with the
\f[V]lazyman\f[R] command:
.PP
\f[V]-4\f[R] : indicates initialize as AstroNvim v4 (use in place of
\f[V]-J\f[R])
.PP
\f[V]-9\f[R] : indicates do not apply any patches to this configuration
.PP
\f[V]-A\f[R] : indicates install all supported Neovim configurations
.PP
\f[V]-a\f[R] : indicates install and initialize the the
\[aq]AstroNvimPlus\[aq] Neovim configuration
.PP
\f[V]-B\f[R] : indicates install and initialize all \[aq]Base\[aq]
Neovim configurations
.PP
\f[V]-b branch\f[R] : specifies an nvim-lazyman git branch to checkout
.PP
\f[V]-c\f[R] : indicates install and initialize the \[aq]NvChad\[aq]
Neovim configuration
.PP
\f[V]-d\f[R] : indicates debug mode
.PP
\f[V]-D subdir\f[R] : specifies the subdirectory of the repository given
with \f[V]-C url\f[R] to retrieve
.PP
\f[V]-e\f[R] : indicates install and initialize the \[aq]Ecovim\[aq]
Neovim configuration
.PP
\f[V]-E \[aq]config\[aq]\f[R] : execute \[aq]nvim\[aq] with
\[aq]config\[aq] Neovim configuration where \[aq]config\[aq] can be one
of \[aq]lazyman\[aq], \[aq]astronvim\[aq], \[aq]ecovim\[aq],
\[aq]kickstart\[aq], \[aq]lazyvim\[aq], \[aq]lunarvim\[aq],
\[aq]magicvim\[aq], \[aq]spacevim\[aq], or any Neovim configuration
directory in \[aq]\[ti]/.config\[aq].
For example, \[aq]lazyman -E lazyvim foo.lua\[aq] would edit
\[aq]foo.lua\[aq] with the LazyVim config
.PP
\f[V]-f \[aq]path\[aq]\f[R] : fix treesitter \[aq]help\[aq] parser in
config file \[aq]path\[aq]
.PP
\f[V]-F \[aq]menu\[aq]\f[R] : indicates present the specified Lazyman
menu where \[aq]menu\[aq] can be one of \[aq]main\[aq], \[aq]conf\[aq],
\[aq]lsp\[aq], \[aq]format\[aq], \[aq]plugin\[aq], \[aq]lazyide\[aq], or
\[aq]webdev\[aq]
.PP
\f[V]-G\f[R] : indicates no plugin manager, initialize with
\f[V]:TSUpdate\f[R]
.PP
\f[V]-g\f[R] : indicates install and initialize the \[aq]Abstract\[aq]
Neovim configuration
.PP
\f[V]-h\f[R] : indicates use Homebrew rather than the native package
manager (Pacman always used on Arch Linux, Homebrew on macOS)
.PP
\f[V]-H\f[R] : indicates compile and install the nightly Neovim build
.PP
\f[V]-j\f[R] : indicates install and initialize the \[aq]BasicIde\[aq]
Neovim configuration
.PP
\f[V]-k\f[R] : indicates install and initialize the \[aq]Kickstart\[aq]
Neovim configuration
.PP
\f[V]-K conf\f[R] : indicates install and initialize the unsupported
in-development configuration \[aq]conf\[aq]
.PP
\f[V]-l\f[R] : indicates install and initialize the \[aq]LazyVim\[aq]
Neovim configuration
.PP
\f[V]-m\f[R] : indicates install and initialize \[aq]MagicVim\[aq]
Neovim configuration
.PP
\f[V]-M\f[R] : indicates install and initialize \[aq]Mini\[aq] Neovim
configuration
.PP
\f[V]-s\f[R] : indicates install and initialize the \[aq]SpaceVim\[aq]
Neovim configuration
.PP
\f[V]-S\f[R] : indicates show Neovim configuration fuzzy selector menu
.PP
\f[V]-t\f[R] : indicates list all installed Lazyman Neovim
configurations
.PP
\f[V]-T\f[R] : indicates list all uninstalled Lazyman Neovim
configurations
.PP
\f[V]-v\f[R] : indicates install and initialize the \[aq]LunarVim\[aq]
Neovim configuration
.PP
\f[V]-i group\f[R] : specifies a group to install/remove/update.
\[aq]group\[aq] can be one of: astronvim kickstart lazyvim lunarvim
nvchad packer plug
.PP
\f[V]-I\f[R] : indicates install all language servers and tools for
coding diagnostics
.PP
\f[V]-J\f[R] : indicates install indicated repo as an AstroNvim v3
custom configuration
.PP
\f[V]-L lang\f[R] : indicates install the \f[V]lang\f[R] Language
configuration where \[aq]lang\[aq] can be one of: All AlanVim Allaman
CatNvim Cpp Go Go2one Insis Knvim LaTeX LazyIde LunarIde LvimIde Magidc
Nv NV-IDE Orange Python Rust SaleVim Shuvro Webdev
.PP
\f[V]-n\f[R] : indicates dry run, don\[aq]t actually do anything, just
printf\[aq]s
.PP
\f[V]-o\f[R] : indicates input required during initialization
.PP
\f[V]-O name\f[R] : indicates set Lazyman configuration to namespace
\[aq]name\[aq] where \[aq]name\[aq] can be one of \f[V]ecovim\f[R],
\f[V]free\f[R], or \[ga]onno
.PP
\f[V]-P\f[R] : indicates use Packer rather than Lazy to initialize
.PP
\f[V]-p\f[R] : indicates use Plug rather than Lazy to initialize
.PP
\f[V]-q\f[R] : indicates quiet install
.PP
\f[V]-Q\f[R] : indicates exit after performing specified action(s)
.PP
\f[V]-r\f[R] : indicates remove the previously installed configuration
.PP
\f[V]-R\f[R] : indicates remove previously installed configuration and
backups
.PP
\f[V]-C url\f[R] : specifies a URL to a Neovim configuration git
repository
.PP
\f[V]-N nvimdir\f[R] : specifies the folder name to use for the config
given by -C
.PP
\f[V]-U\f[R] : indicates update an existing configuration
.PP
\f[V]-V url\f[R] : specifies an NvChad user configuration git repository
.PP
\f[V]-w conf\f[R] : indicates install and initialize Personal
\[aq]conf\[aq] configuration.
\[aq]conf\[aq] can be one of: All Adib Artur Beethoven Brain Charles
Craftzdog Dillon Elianiva Elijah Enrique Kristijan Heiker J4de Josean
JustinNvim Kodo LamarVim LvimAdib Maddison Metis Mini ONNO OnMyWay
Optixal Orhun Primeagen Rafi RNvim Roiz Simple Slydragonn Spider Traap
Wuelner xero Xiao
.PP
\f[V]-W\f[R] : indicates install and initialize all \[aq]Personal\[aq]
Neovim configurations
.PP
\f[V]-x conf\f[R] : indicates install and initialize nvim-starter
\[aq]conf\[aq] configuration.
\[aq]conf\[aq] can be one of: All 2k AstroNvimStart Barebones Basic
CodeArt Cosmic Ember Fennel HardHacker JustinOhMy Kabin Kickstart Micah
Normal NvPak Modern pde Rohit Scratch SingleFile BasicLsp BasicMason
Extralight LspCmp Minimal StartBase Opinion StartLsp StartMason Modular
.PP
\f[V]-X\f[R] : indicates install and initialize all \[aq]Starter\[aq]
Neovim configurations
.PP
\f[V]-y\f[R] : indicates do not prompt, answer \[aq]yes\[aq] to any
prompt
.PP
\f[V]-Y\f[R] : indicates use the following arguments as
\[aq]name\[aq]/\[aq]value\[aq] to set Lazyman config.
For example: \f[V]lazyman -Y theme kanagawa\f[R].
If the \[aq]name\[aq] argument is \[aq]get\[aq] then the current value
is returned
.PP
\f[V]-z\f[R] : indicates do not run nvim after initialization
.PP
\f[V]-Z\f[R] : indicates do not install Homebrew, Neovim, or any other
tools during initialization
.PP
\f[V]-u\f[R] : displays this usage message and exits
.PP
\f[V]health\f[R] : generate and display a health check for a
configuration
.PP
\f[V]info\f[R] : open an information page for a configuration in the
default browser
.PP
\f[V]init\f[R] : initialize specified Neovim configuration and exit
.PP
\f[V]install\f[R] : fuzzy search and select configuration to install
.PP
\f[V]install bob\f[R] : install the Bob Neovim version manager
.PP
\f[V]open\f[R] : fuzzy search and select configuration to open
.PP
\f[V]remove\f[R] : fuzzy search and select configuration to remove
.PP
\f[V]search\f[R] : fuzzy search and select configurations for a plugin
.PP
\f[V]status\f[R] : displays a brief status report and exits
.PP
\f[V]usage\f[R] : displays a usage message and exits
.PP
Commands act on \f[V]NVIM_APPNAME\f[R], override with \[aq]-N
nvimdir\[aq] or \[aq]-A\[aq]
.PP
Without arguments lazyman installs and initializes nvim-Lazyman or, if
initialized presents an interactive menu system.
.PP
See the Lazyman website Usage section (https://lazyman.dev/usage) for
additional information on Lazyman usage.
.SS EXAMPLES
.PP
\f[V]$HOME/.config/nvim-Lazyman/lazyman.sh\f[R] : initializes the
\f[V]Lazyman\f[R] Neovim configuration in
\f[V]$HOME/.config/nvim-Lazyman/\f[R]
.PP
\f[V]lazyman\f[R] : presents an interactive menu interface
.PP
\f[V]lazyman install\f[R] : fuzzy search and select a configuration to
install and initialize
.PP
\f[V]lazyman open\f[R] : fuzzy search and select an initialized
configuration to open
.PP
\f[V]lazyman -a\f[R] : installs and initializes the
\f[V]AstroNvimPlus\f[R] Neovim configuration in
\f[V]$HOME/.config/nvim-AstroNvimPlus/\f[R]
.PP
\f[V]lazyman -c\f[R] : installs and initializes the \f[V]NvChad\f[R]
Neovim configuration in \f[V]$HOME/.config/nvim-NvChad/\f[R]
.PP
\f[V]lazyman -l\f[R] : installs and initializes the \f[V]LazyVim\f[R]
Neovim configuration in \f[V]$HOME/.config/nvim-LazyVim/\f[R]
.PP
\f[V]lazyman -A\f[R] : installs and initializes all supported Neovim
configurations
.PP
\f[V]lazyman -I\f[R] : installs language servers and tools for coding
diagnostics
.PP
\f[V]lazyman -U -N nvim-LazyVim\f[R] : updates the \f[V]LazyVim\f[R]
Neovim configuration in \f[V]$HOME/.config/nvim-LazyVim/\f[R]
.PP
\f[V]lazyman -A -U\f[R] : updates all installed supported Neovim
configurations
.PP
\f[V]lazyman -P -C https://github.com/Abstract-IDE/Abstract -N nvim-Abstract\f[R]
: installs and initializes the Packer based \[aq]Abstract\[aq] Neovim
configuration in \f[V]\[ti]/.config/nvim-Abstract\f[R]
.PP
\f[V]lazyman -R -N nvim-LazyVim\f[R] : removes the \f[V]LazyVim\f[R]
Neovim configuration in \f[V]$HOME/.config/nvim-LazyVim/\f[R], its data
files, cache, state, and all backups
.PP
Sometimes people place their Neovim configuration in a repository
subdirectory along with other configurations in a \f[V]dotfiles\f[R]
repo.
To retrieve only the Neovim configuration subdirectory in such a
repository, use the \f[V]-b branch\f[R] and \f[V]-D subdir\f[R]
arguments to \f[V]lazyman\f[R] along with \f[V]-C url\f[R] and
\f[V]-N nvimdir\f[R].
If no \f[V]-b branch\f[R] is provided then the default git branch is
assumed to be \f[V]master\f[R].
For example, to install and initialize the Neovim configuration hosted

CHANGELOG.md
# Changelog

## Unreleased

### ⚡️ Added

### 🐞 Fixed

## v3.1.6r2

### ⚡️ Added

- Add security policy
- Add `Vimacs` to `Personal`
- Add enable/disable AstroNvim Community plugins to `anvmv4_config.sh`
- Add community modules enable/disable to `AstroNvimV4` plugins menu
- Add `treesitter-context` to `AstroNvimV4` plugins menu
- Update `AstroNvimV4` info documents, remove `astrolsp.lua` from `lazyman` initialization
- Add note about upgrading `git` and `bash`
- Add auto format to `AstroNvimV4` formatters configuration menu
- Add arg processing for `AstroNvimV4` submenus from command line
- Add `cheatsheet` and smooth scrolling plugins config to `AstroNvimV4` config script

### 🐞 Fixed

- Several configurations switched from `Packer` to `Lazy.nvim`
- Fixed checks for Neovim 0.10
- Enable `neovim-session-manager` only if Neovim 0.10 is in use
- Remove `bufferline.nvim` from `onno` namespace
- Disable `read:scope` warning from `octo.nvim`
- Plugin configuration changes for 0.10.0 support
- Do not initialize `Vimacs` during installation
- Use new starter URL for `NvChad` based configs
- Backup original `configuration.lua` for managed configurations during initialization
- Preserve customized `configuration.lua` over updates to `AstroNvimV4`, `Webdev`, and `LazyIde`
- Remove `Daniel` LunarVim configuration from `Personals`
- Fix `shellcheck` errors in `anvmv4_config.sh`
- Remove `java_debug.sh` install script invocation from `lazyman.sh`
- Check for `java` in `java_debug.sh`
- Remove duplicate calls to setup in `neoscroll` config

## v3.1.6r1

### ⚡️ Added

- Customize `LSP` and `Formatters` menus in `AstroNvimV4` configure script
- Customize `Plugins` menu in `AstroNvimV4` configure script
- Add `LSP`, `Formatters`, and `Plugins` menus to `AstroNvimV4` configuration menu script
- Extend `scripts/get_conf.lua` so it works with any `NVIM_APPNAME` not just `Lazyman`
- Add configuration menu script for `AstroNvimV4` and add it to `Lazyman` menus

### 🐞 Fixed

- Fix setup and loading of `monokai-pro` colorscheme
- Update `AstroNvimV4` info documents
- Don't automatically enable `gopls`

## v3.1.5r4

### ⚡️ Added

- Install `stylua` in `install_neovim.sh`
- Add `concat_tables` function, check both lsp tables in `lspconfig`
- Install `taplo` with `mason-lspconfig`
- Use `pip` to install `pynvim`
- Use GNU sed on `macOS`
- Use Homebrew to install `Bob` if available

### 🐞 Fixed

- Use temporary `astrolsp.lua` for `AstroNvimV4` initialization
- Move `jsonls` out of installed lsp servers, move `vimls` in
- Fix update categories, was installing all instead of updating installed

## v3.1.5r3

### ⚡️ Added

- Add `AstroNvimV4` to `starters` in `configrc`
- Use `pipx` on `macOS`

### 🐞 Fixed

- Initialize `AstroNvimV4` twice
- Remove tmp uninstall script after executing
- Reflect renamed `schemastore` repo `b0o/SchemaStore.nvim` in `lspconfig`
- Remove `Neorg` from `free` namespace
- Update `obsidian` plugin configuration

## v3.1.5r2

### ⚡️ Added

- Add `-4` argument to `lazyman` to indicate use `AstroNvim` v4 during initialization
- Add `AstroNvimV4` to `Starter` configurations as example `AstroNvim` v4 configuration

### 🐞 Fixed

## v3.1.5r1

### ⚡️ Added

### 🐞 Fixed

- Pin all `AstroNvim` configuration initializations to `v3.45.3` for now

## v3.1.4r3

### ⚡️ Added

- Install `kitten` in Kitty install script

### 🐞 Fixed

- Remove Allaman patch
- Remove Allaman custom config on removal of Allaman
- Create custom Allaman config if none exists
- Install `gpatch` with `brew` if Homebrew is being used
- No need to install `pip` with `pipx`
- Update `ufo` configuration replacing deprecated `close_fold_kinds`
- Do not enable `gopls` on macOS

## v3.1.4r2

### ⚡️ Added

- Add a couple of configs to check out in TODO
- Add `reportUnusedExpression = "none"` to `pyright` setup

### 🐞 Fixed

- Remove `JustinLvim`, no longer maintained
- Fix Neovide asset download

## v3.1.4r1

### ⚡️ Added

- Support for Ubuntu 23.04 and greater using `pipx` for Python module installation

### 🐞 Fixed

## v3.1.3r5

### ⚡️ Added

- Convenience script `lman` to invoke Neovim with the Lazyman configuration
- Install `pyright`, remove `misspell` tmp files
- Add argument to `install_neovim.sh` to disable `ulimit` call for use in container builds
- Source `nvm.sh` after running `install_neovim.sh`

### 🐞 Fixed

- Ignore empty lines in `.nvimdirs`, toggle `pylsp` and `pyright`
- Remove `nvim-ts-context-commentstring`, use `mini.comment` exclusively

## v3.1.3r3

- Bug fix release

### ⚡️ Added

### 🐞 Fixed

- Install g++ as a Neovim dependency
- Add `$HOME/.local/bin` to PATH when first installing Neovim in `lazyman.sh`

## v3.1.3r2

### ⚡️ Added

- Use `fzf` and `lsd` to provide a preview in `nvims` shell function
- Install latest `lsd` release, check architecture for deb release downloads
- Add note about Kasm registry and Neovim Kasm workspace
- Add `wing` to `treesitter` installed parsers
- Add note about rate limiting, add `GITHUB_TOKEN` as env var for Github API requests
- Use Github API token for Github API requests if one exists
- Add notice to be patient when installing Neovim dependencies
- Add `noinstall` argument to `lazyman` and `-s` to `install_neovim.sh` to specify no native installation package installs
- Add support for `nvim-treesitter-context`
- Add support for `SeniorMars` personal config
- Disable LSP timeout by default, not yet ready for primetime
- Add support for LSP timeout plugin

### 🐞 Fixed

- Add patch for `SeniorMars` config
- Check for nil rhs in lsp keymaps
- Add newline if brief usage exits
- Fix install neovide script with new download filenames
- Disable LSP timeout by default
- Pin `indent-blankline` to version 2.20.8

## v3.1.3r1

### ⚡️ Added

- Split installed LSP servers out into separate config
- Update servers and formatters configured in LSP menu
- Merge `ecovim` `nvim-tree` config into `config/nvim-tree.lua`
- Update `nivm-tree` opts

lua/free/config/startup/lazyman.lua
local datetime = os.date("  %Y-%b-%d   %H:%M:%S", os.time())
local version = vim.version()
local version_info = ""
if version ~= nil then
  version_info = "v" .. version.major .. "." .. version.minor .. "." .. version.patch
end

local stats = require("lazy").stats()
local vinfo = "Neovim " .. version_info
local lazystats = "  loaded " .. stats.count .. " plugins "
if vim.o.filetype == "lazy" then
  vim.cmd.close()
end

local theme_switch = "Telescope colorscheme"
if settings.enable_telescope_themes then
  theme_switch = "Telescope themes"
end
local settings = {
  header = {
    type = "text",
    oldfiles_directory = false,
    align = "center",
    fold_section = false,
    title = "Header",
    margin = 5,
    content = {
      "  _                                            ",
      " | |                                           ",
      " | |  __,   __         _  _  _    __,   _  _   ",
      " |/  /  |  / / _|   | / |/ |/ |  /  |  / |/ |  ",
      " |__/\\_/|_/ /_/  \\_/|/  |  |  |_/\\_/|_/  |  |_/",
      "             /|    /|                          ",
      "             \\|    \\|                          ",
    },
    highlight = "Statement",
    default_color = "",
    oldfiles_amount = 0,
  },
  body = {
    type = "mapping",
    oldfiles_directory = false,
    align = "center",
    fold_section = false,
    title = "Lazyman Commands",
    margin = 5,
    content = {
      { " New File",          "lua require'startup'.new_file()",        "n" },
      { " Find File",         "Telescope find_files",                   "f" },
      { " Find Word",         "Telescope live_grep",                    "w" },
      { " Recent Files",      "Telescope oldfiles prompt_title=Recent", "r" },
      { " File Browser",      "Telescope file_browser",                 "b" },
      { " Search Zoxide",     "Telescope zoxide list",                  "z" },
      { " Colorschemes",      theme_switch,                             "s" },
      { " Cheatsheet",        "Cheatsheet",                             "c" },
      { " Neovim Health",     "checkhealth",                            "h" },
      { " Lazyman Main Menu", "Lazyman",                                "l" },
      { " Lazyman Conf Menu", "Lazyconf",                               "C" },
      { " Manage Plugins",    "Lazy",                                   "L" },
      { " Manage Packages",   "Mason",                                  "M" },
      { " Quit",              "quitall",                                "q" },
    },
    highlight = "String",
    default_color = "",
    oldfiles_amount = 0,
  },
  footer = {
    type = "text",
    oldfiles_directory = false,
    align = "center",
    fold_section = false,
    title = "Footer",
    margin = 5,
    content = { datetime .. "  " .. vinfo .. lazystats },
    highlight = "Number",
    default_color = "",
    oldfiles_amount = 0,
  },
  options = {
    mapping_keys = true,
    cursor_column = 0.5,
    empty_lines_between_mappings = false,
    disable_statuslines = true,
    paddings = { 0, 1, 1, 0 },
  },
  mappings = {
    execute_command = "<CR>",
    open_file = "o",
    open_file_split = "<c-o>",
    open_section = "<TAB>",
    open_help = "?",
  },
  colors = {
    background = "#1f2227",
    folded_section = "#56b6c2",
  },
  parts = { "header", "body", "footer" },
}
return settings
